modules/bridge/src/main/ts/ephox/bridge/components/menu/Description.ts

```diff
@@ -0,0 +1,25 @@
+import { FieldSchema, ValueSchema } from '@ephox/boulder';
+import { Result } from '@ephox/katamari';
+
+export interface DescriptionSpec {
+  type: 'description';
+  text: string;
+  classes?: string[];
```

Comment:
Should we have classes here? I thought we were specifically avoiding allowing integrators to use classes, as it caused support issues.

Reply:
We had a "Quality of Life Improvements" discussion about this and we considered two alternatives when it comes to styling: 1) add style attributes for a set of safe styles or 2) allow classes and document what kind of safe styles (non-layout) that an integrator may use. We thought alternative 2 would be the better choice here. Anything you would like to add @lostkeys or @metricjs?

Reply:
If we're going to do that it needs to be a separate PR and not something partially done here.

Reply:
@lnewson We decided classes on these elements is a reasonable compromize between a very verbose list of attributes props and allowing the integrator to fully custom html and css. Since the class is applied to know elements and in a structured way, we can much better anticipate what integrators might do than if they could throw any complex HTML Dom tree inside our UI. Its mainly changes our reset and layout css that could break customers integrations, but when targeted so specifically as we do here, it's not as "dangerous".

Reply:
That might be true now, but what about in future if we add more components? I'm probably okay with adding it on the `container`, but it shouldn't be on `image`, `title` or `description` as those could end up being used elsewhere potentially. It'd also be. very confusing to users to have somethings that allow classes, but other things that don't.

Reply:
Sorry but it's the container we _don't_ want users to add classes to. It was unfortunate you weren't in the meeting. The reason for this is to be able to style all the things needed for the flexibility of this menu item. Such as setting width, height and alignment of images, setting font sizes, colors, text transforms etc on text nodes. These things cannot go in the skin as we must support different styles for different autocompleter menus, as well as cases where a menu item can contains multiple images. The other solution was to use a styles property but we deemed classes be the more robust option. 

Here is an image showing an example what we want to be able to do
![Untitled-2](https://user-images.githubusercontent.com/3122929/97860797-11763900-1d03-11eb-87ab-183f9bdc7e98.png)

Reply:
Fair enough, in that case though then the new components shouldn't be generic and need to be more specific, as we don't want `classes` on something generic.

Reply:
To clarify, when I say more specific, I mean calling them `cardtitle` instead of `title` for example.

Reply:
Ok I (think I) see, I have no problem with renaming the component to be `cardtext` or something similar to avoid people thinking they can add classes to any `title` property they find in TinyMCE. That makes total sense.

Reply:
Yeah, that's all I meant if we want to allow classes to be applied for this use case 👍

Reply:
I have renamed the components to `cardtext` and `cardimage` (Description has been removed)